File: Java/GenericStructures/Stack.java
package GenericStructures;

public class Stack <T>{
    private class Node{
        T data; 
        Node next = null;
        Node(T data){
            this.data = data;
        }
    }
    
    private int size;
    private Node top = null;
    // private Node bottom = null; -- implement this 

    public Stack(){
        this.size = 0;
    }

    public void push(T data){
        Node node = new Node(data);
        node.next = this.top;
        this.top = node;
        ++ this.size;
    }

    public void pop(){
        try {
            if(this.size == 0) throw new Exception();
            this.top = top.next;
            -- this.size;

        } catch (Exception e) {
            System.out.println("Error: Stack is empty.");
            return;
        }
    }
    
    public int size(){
        return this.size;
    }

    public T top(){
        return this.top.data;
    }
}

class MainStack{
    public static void main(String[] args){

    }
}

File: Python/GenericStructures/Stack.py
from typing import TypeVar, Generic

T = TypeVar("T")


class Node(Generic[T]):
    next: Node[T] = None

    def __init__(self, data: T) -> None:
        self.data = data


class Stack(Generic[T]):
    top: Node[T] = None
    size: int = 0

    def push(self, data: int) -> None:
        node = Node[T](data)
        node.next = self.top
        self.top = node
        self.size += 1

    def pop(self) -> None:
        if self.is_empty():
            raise Exception("Stack is already")
        self.top = self.top.next
        self.size -= 1

    def display(self) -> None:
        ptr = self.top
        while ptr.next != None:
            print(ptr.data)
            ptr = ptr.next

    def is_empty(self) -> bool:
        return self.size == 0


if __name__ == "__main__":
    s = Stack()
    for i in range(0, 9):
        s.push(i)

    s.pop()
    s.display()
